client/rest-high-level/src/main/java/org/elasticsearch/client/response/indices/flush/SyncedFlushResponse.java

```diff
@@ -0,0 +1,329 @@
+                    other.successful == this.successful &&
+                    other.failed == this.failed;
+            } else {
+                return false;
+            }
+        }
+
+    }
+
+    public static final class IndexResult implements ToXContentFragment {
+
+        // Field name declaration
+        public static final String FAILURES = "failures";
+        // End field name declaration
+
+        private ShardCounts counts;
+        private List<ShardFailure> failures;
+
+        public IndexResult(ShardCounts counts, List<ShardFailure> failures) {
+            this.counts = new ShardCounts(counts.total, counts.successful, counts.failed);
+            this.failures = Collections.unmodifiableList(failures);
+        }
+
+        public int totalShards() {
+            return counts.total;
+        }
+
+        public int successfulShards() {
+            return counts.successful;
+        }
+
+        public int failedShards() {
+            return counts.failed;
+        }
+
+        public List<ShardFailure> failures() {
+            return failures;
+        }
+
+        protected ShardCounts getShardCounts() {
+            return counts;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            counts.toXContent(builder, params);
+            if (failures.size() > 0) {
+                builder.startArray(FAILURES);
+                for (ShardFailure failure : failures) {
+                    failure.toXContent(builder, params);
+                }
+                builder.endArray();
+            }
+            return builder;
+        }
+
+        public static IndexResult fromXContent(XContentParser parser) throws IOException {
+            ensureExpectedToken(Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+            ShardCounts counts = ShardCounts.fromXContent(parser);
+            List<ShardFailure> failures = new ArrayList<>();
+            while (parser.nextToken().equals(Token.FIELD_NAME)) {
+                if (parser.currentName().equals(FAILURES)) {
+                    ensureExpectedToken(Token.START_ARRAY, parser.nextToken(), parser::getTokenLocation);
+                    while (parser.nextToken().equals(Token.START_OBJECT)) {
+                        failures.add(ShardFailure.fromXContent(parser));
+                    }
+                    // We should be at the end of the array by now
+                    ensureExpectedToken(Token.END_ARRAY, parser.currentToken(), parser::getTokenLocation);
+                }
+            }
+            // We should be at the end of the index object by now
+            ensureExpectedToken(Token.END_OBJECT, parser.currentToken(), parser::getTokenLocation);
+            return new IndexResult(counts, failures);
+        }
+    }
+
+    // Only used as a container for XContent
+    public static final class ShardFailure implements ToXContentFragment {
```

Comment:
I don't think the comment above is accurate. It looks like you return it from the API so it deserves some description like "Description of a shard that failed to synced flush".